infra/storage/file_transfer_managers/s3Mgr.class.php

```diff
@@ -79,14 +80,22 @@ protected function __construct(array $options = null)
 		{
 			$this->storageClass = $options['storageClass'];
 		}
-
-		if (class_exists('KBatchBase'))
+		
+		if($options && isset($options['s3Arn']))
 		{
-			$this->s3Arn = kBatchUtils::getKconfParam('arnRole', true);
+			$this->s3Arn = $options['s3Arn'];
+			$this->hash = true;
```

Comment:
Why no simply save the hash here instead of true? 
Also why is this the only case where we hash, what about the other case? I think you should hash there as well to be able to distinguish between creds used by the file transfer manager and the sharedFileSystemWrtappers.